man/hsize.1
.ie \n(.g .ds Aq \(aq
.el .ds Aq '
.TH hsize 1  "hsize 0.1.0" 
.SH NAME
hsize \- Convert file sizes to and from human\-readable units
.SH SYNOPSIS
\fBhsize\fR [\fB\-p\fR|\fB\-\-precision\fR] [\fB\-f\fR|\fB\-\-from\-scale\fR] [\fB\-B\fR|\fB\-\-from\-binary\fR] [\fB\-t\fR|\fB\-\-to\-scale\fR] [\fB\-b\fR|\fB\-\-to\-binary\fR] [\fB\-e\fR|\fB\-\-scientific\-notation\fR] [\fB\-s\fR|\fB\-\-separator\fR] [\fB\-n\fR|\fB\-\-no\-b\-suffix\fR] [\fB\-S\fR|\fB\-\-skip\-short\-numbers\fR] [\fB\-h\fR|\fB\-\-help\fR] [\fB\-V\fR|\fB\-\-version\fR] [\fISIZES\fR] [\fIsubcommands\fR]
.SH DESCRIPTION
Convert file sizes to and from human\-readable units
.SH OPTIONS
.TP
\fB\-p\fR, \fB\-\-precision\fR \fI<PRECISION>\fR [default: 1]
Number of decimal places to display in the converted number
.TP
\fB\-f\fR, \fB\-\-from\-scale\fR \fI<SCALE>\fR
Size scale of the given numbers
.br

.br
\fIPossible values:\fR
.RS 14
.IP \(bu 2
b
.IP \(bu 2
k
.IP \(bu 2
m
.IP \(bu 2
g
.IP \(bu 2
t
.IP \(bu 2
p
.IP \(bu 2
e
.IP \(bu 2
z
.IP \(bu 2
y
.IP \(bu 2
r
.IP \(bu 2
q
.RE
.TP
\fB\-B\fR, \fB\-\-from\-binary\fR
Given numbers are powers of 2 (1K = 1024)
.TP
\fB\-t\fR, \fB\-\-to\-scale\fR \fI<SCALE>\fR
Size scale of the converted numbers
.br

.br
\fIPossible values:\fR
.RS 14
.IP \(bu 2
b
.IP \(bu 2
k
.IP \(bu 2
m
.IP \(bu 2
g
.IP \(bu 2
t
.IP \(bu 2
p
.IP \(bu 2
e
.IP \(bu 2
z
.IP \(bu 2
y
.IP \(bu 2
r
.IP \(bu 2
q
.RE
.TP
\fB\-b\fR, \fB\-\-to\-binary\fR
Converted numbers should be powers of 2 (1K = 1024)
.TP
\fB\-e\fR, \fB\-\-scientific\-notation\fR
Numbers should be displayed in scientific notation
.TP
\fB\-s\fR, \fB\-\-separator\fR \fI<SEPARATOR>\fR [default:  ]
Characters to put between the number and unit
.TP
\fB\-n\fR, \fB\-\-no\-b\-suffix\fR
Remove the \*(AqB\*(Aq from the end of the unit (MB \-> M)
.TP
\fB\-S\fR, \fB\-\-skip\-short\-numbers\fR
Skip converting numbers if they\*(Aqll end up being longer than the original
.TP
\fB\-h\fR, \fB\-\-help\fR
Print help
.TP
\fB\-V\fR, \fB\-\-version\fR
Print version
.TP
[\fISIZES\fR]

.SH SUBCOMMANDS
.TP
hsize\-replace(1)
Use regex to search and replace numbers
.TP
hsize\-generate(1)
Generate various shell command files
.SH VERSION
v0.1.0


completions/hsize.bash
_hsize() {
    local i cur prev opts cmd
    COMPREPLY=()
    if [[ "${BASH_VERSINFO[0]}" -ge 4 ]]; then
        cur="$2"
    else
        cur="${COMP_WORDS[COMP_CWORD]}"
    fi
    prev="$3"
    cmd=""
    opts=""

    for i in "${COMP_WORDS[@]:0:COMP_CWORD}"
    do
        case "${cmd},${i}" in
            ",$1")
                cmd="hsize"
                ;;
            hsize,g)
                cmd="hsize__generate"
                ;;
            hsize,gen)
                cmd="hsize__generate"
                ;;
            hsize,generate)
                cmd="hsize__generate"
                ;;
            hsize,help)
                cmd="hsize__help"
                ;;
            hsize,r)
                cmd="hsize__replace"
                ;;
            hsize,re)
                cmd="hsize__replace"
                ;;
            hsize,replace)
                cmd="hsize__replace"
                ;;
            hsize__generate,c)
                cmd="hsize__generate__completions"
                ;;
            hsize__generate,comp)
                cmd="hsize__generate__completions"
                ;;
            hsize__generate,completions)
                cmd="hsize__generate__completions"
                ;;
            hsize__generate,help)
                cmd="hsize__generate__help"
                ;;
            hsize__generate,m)
                cmd="hsize__generate__manpages"
                ;;
            hsize__generate,man)
                cmd="hsize__generate__manpages"
                ;;
            hsize__generate,manpages)
                cmd="hsize__generate__manpages"
                ;;
            hsize__generate__help,completions)
                cmd="hsize__generate__help__completions"
                ;;
            hsize__generate__help,help)
                cmd="hsize__generate__help__help"
                ;;
            hsize__generate__help,manpages)
                cmd="hsize__generate__help__manpages"
                ;;
            hsize__help,generate)
                cmd="hsize__help__generate"
                ;;
            hsize__help,help)
                cmd="hsize__help__help"
                ;;
            hsize__help,replace)
                cmd="hsize__help__replace"
                ;;
            hsize__help__generate,completions)
                cmd="hsize__help__generate__completions"
                ;;
            hsize__help__generate,manpages)
                cmd="hsize__help__generate__manpages"
                ;;
            *)
                ;;
        esac
    done

    case "${cmd}" in
        hsize)
            opts="-p -f -B -t -b -e -s -n -S -h -V --precision --from-scale --from-binary --to-scale --to-binary --scientific-notation --separator --no-b-suffix --skip-short-numbers --help --version [SIZES]... replace r re generate g gen help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 1 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --precision)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -p)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --from-scale)
                    COMPREPLY=($(compgen -W "b k m g t p e z y r q" -- "${cur}"))
                    return 0
                    ;;
                -f)
                    COMPREPLY=($(compgen -W "b k m g t p e z y r q" -- "${cur}"))
                    return 0
                    ;;
                --to-scale)
                    COMPREPLY=($(compgen -W "b k m g t p e z y r q" -- "${cur}"))
                    return 0
                    ;;
                -t)
                    COMPREPLY=($(compgen -W "b k m g t p e z y r q" -- "${cur}"))
                    return 0
                    ;;
                --separator)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -s)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        hsize__generate)
            opts="-h --help completions c comp manpages m man help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 2 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        hsize__generate__completions)
            opts="-s -h --shell --help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --shell)
                    COMPREPLY=($(compgen -W "bash elvish fish powershell zsh" -- "${cur}"))
                    return 0
                    ;;
                -s)
                    COMPREPLY=($(compgen -W "bash elvish fish powershell zsh" -- "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        hsize__generate__help)
            opts="completions manpages help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        hsize__generate__help__completions)
            opts=""
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        hsize__generate__help__help)
            opts=""
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        hsize__generate__help__manpages)
            opts=""
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        hsize__generate__manpages)
            opts="-o -h --output-directory --help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --output-directory)
                    COMPREPLY=()
                    if [[ "${BASH_VERSINFO[0]}" -ge 4 ]]; then
                        compopt -o plusdirs
                    fi
                    return 0
                    ;;
                -o)
                    COMPREPLY=()
                    if [[ "${BASH_VERSINFO[0]}" -ge 4 ]]; then
                        compopt -o plusdirs
                    fi
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        hsize__help)
            opts="replace generate help"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 2 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        hsize__help__generate)
            opts="completions manpages"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        hsize__help__generate__completions)
            opts=""
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        hsize__help__generate__manpages)
            opts=""
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 4 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        hsize__help__help)
            opts=""
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        hsize__help__replace)
            opts=""
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 3 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
        hsize__replace)
            opts="-r -U -L -i -h --regex --multi-line --left-align --in-place --help [FILES]..."
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 2 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --regex)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -r)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
    esac
}

if [[ "${BASH_VERSINFO[0]}" -eq 4 && "${BASH_VERSINFO[1]}" -ge 4 || "${BASH_VERSINFO[0]}" -gt 4 ]]; then
    complete -F _hsize -o nosort -o bashdefault -o default hsize
else
    complete -F _hsize -o bashdefault -o default hsize
fi
